integrations/FacebookPixel/browser.js

```diff
@@ -120,6 +128,10 @@ class FacebookPixel {
     if (properties) {
       const { revenue, currency } = properties;
       revValue = this.formatRevenue(revenue);
+      if (!isDefined(revValue)) {
```

Comment:
@utsabc This method has become too big and risky to maintain. Moreover, some snippets are repeated.
I observed at least two mistakes during this review, which I attribute to the above.